sdk/src/Handlers/EntityFramework/EFUtil.cs

```diff
@@ -0,0 +1,122 @@
+    /// Utilities for EFInterceptor
+    /// </summary>
+    public static class EFUtil
+    {
+        private static readonly string DefaultDatabaseType = "EntityFrameworkCore";
+
+        private static readonly string[] UserIdFormatOptions = { "user id", "username", "user" }; // case insensitive
+
+        // Some database providers may not support Entity Framework Core 3.0 and above now
+        // https://docs.microsoft.com/en-us/ef/core/providers/?tabs=dotnet-core-cli
+        private static readonly Dictionary<string, string> DatabaseTypes = new Dictionary<string, string>()
+        {
+            // Support tracing
+            { "Microsoft.EntityFrameworkCore.SqlServer" , "sqlserver" },
+            { "Microsoft.EntityFrameworkCore.Sqlite" , "sqlite" },
+            { "Npgsql.EntityFrameworkCore.PostgreSQL" , "postgresql" },
+            { "Pomelo.EntityFrameworkCore.MySql" , "mysql" },
+            { "FirebirdSql.EntityFrameworkCore.Firebird" , "firebirdsql" },
+
+            // Not support tracing so far
+            { "Microsoft.EntityFrameworkCore.InMemory" , "inmemory" },
+            { "Microsoft.EntityFrameworkCore.Cosmos" , "cosmosdb" },
+            { "Devart.Data.MySql.EFCore" , "mysql" },
+            { "Devart.Data.Oracle.EFCore" , "oracle" },
+            { "Devart.Data.PostgreSql.EFCore" , "postgresql" },
+            { "Devart.Data.SQLite.EFCore" , "sqlite" },
+            { "FileContextCore" , "filecontextcore" },
+            { "EntityFrameworkCore.Jet" , "jet" },
+            { "EntityFrameworkCore.SqlServerCompact35" , "sqlservercompact35" },
+            { "EntityFrameworkCore.SqlServerCompact40" , "sqlservercompact40" },
+            { "Teradata.EntityFrameworkCore" , "teradata" },
+            { "EntityFrameworkCore.FirebirdSql" , "firebirdsql" },
+            { "EntityFrameworkCore.OpenEdge" , "openedge" },
+            { "MySql.Data.EntityFrameworkCore" , "mysql" },
+            { "Oracle.EntityFrameworkCore" , "oracle" },
+            { "IBM.EntityFrameworkCore" , "ibm" },
+            { "IBM.EntityFrameworkCore-lnx" , "ibm" },
+            { "IBM.EntityFrameworkCore-osx" , "ibm" },
+            { "Pomelo.EntityFrameworkCore.MyCat" , "mycat" }
+        };
+
+        private static readonly Regex _portNumberRegex = new Regex(@"[,|:]\d+$");
+
+        /// <summary>
+        /// Extract database_type from <see cref="DbContext"/>.
+        /// </summary>
+        /// <param name="context">Instance of <see cref="DbContext"/>.</param>
+        /// <returns>Type of database.</returns>
+        public static string GetDataBaseType(DbContext context)
+        {
+            string databaseProvider = context?.Database?.ProviderName;
+
+            // Need to check if the context and its following parameter is null or not to avoid exception
+            if (string.IsNullOrEmpty(databaseProvider))
+            {
+                return DefaultDatabaseType;
+            }
+
+            string value = null;
```

Comment:
If the `value` is not found, then this method will return the database provider name, which will not cause NPEs.

Reply:
> I would recommend initializing `value` to an empty string to avoid NPEs down the line.

If the value is not found, then this method will return the database provider name, which will not cause NPEs.